DataAccess/Public/Include/DataAccess/AbstractRepositoryManager.hpp

```diff
@@ -8,158 +8,148 @@
 #include <type_traits>
 
 #include "AbstractRepositoryContainer.hpp"
-#include "IServerRepositories.hpp"
 #include "RepositoryRequest.hpp"
 
-#include <DataAccess.Postgre/PostgreRepositories.hpp>
 
 namespace DataAccess
 {
-    template <typename TIRepository, typename TReturn, typename... TArgs>
-    using MethodReference = TReturn(TIRepository::*)(TArgs...);
+template <typename TIRepository, typename TReturn, typename... TArgs>
+using MethodReference = TReturn (TIRepository::*)(TArgs...);
+
+/**
+ * @brief fmt function uses for converting arguments, that you pass to repository method to correct form.
+ * @param ref argument that is passing to repository method.
+ * @return Return correct format argument: for fundamental types like int, float, etc - by value, all other - by const ref.
+ */
+template <typename Type>
+std::conditional_t<std::is_fundamental_v<Type>, std::remove_reference_t<Type>, const Type&> fmt(const Type& ref)
+{
+    return ref;
+}
+
+/**
+ * @class IRepositoryManager
+ * @brief IRepositoryManager is template class, which realization controls handler for repository requests.
+ * @details Class controls pushing to queue and further processing of requests (have own thread for it).
+ * Requests with the highest priority are at the top of the queue.
+ */
+class IRepositoryManager
+{
+protected:
+    std::unique_ptr<AbstractRepositoryContainer> _repositories;
+    std::priority_queue<RepositoryRequest>       _queue;
+
+    std::mutex              _queueMutex;
+    std::condition_variable _queueCV;
+
+    std::atomic<bool> _handlerState = true;
+    std::thread       _repositoryRequestsHandler;
+
+public:
+    IRepositoryManager() {}
```

Comment:
We should make this constructor the default, since we're not going to move it anywhere, as I understand, when we have the move constructor removed, as well as the copy constructor.